tests/SAML2/StatusResponseTest.php

```diff
@@ -18,12 +22,21 @@ class StatusResponseTest extends \PHPUnit\Framework\TestCase
      */
     public function testMarshalling(): void
     {
+        $status = new Status();
+
+        $subcode = new StatusCode();
+        $subcode->setValue('OurSubStatusCode');
+        $code = new StatusCode();
+        $code->setValue('OurStatusCode');
+        $code->setSubCodes([$subcode]);
+        $status->setStatusCode($code);
+
+        $message = new StatusMessage();
+        $message->setMessage('OurMessageText');
+        $status->setStatusMessage($message);
```

Comment:
With an API inverted to work the way we are discussing, this could be simplified to:

```php
        $status = new Status(
            new StatusCode(
                'OurStatusCode',
                [new StatusCode('OurSubStatusCode')]
            ),
            new StatusMessage('OurMessageText')
        ),
```

Is that easier to read?